File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    answer_arr = []
    for i in numbers:
        list_of_nums = list(str(i))
        result = 0
        for j in list_of_nums:
            power_of = int(j) ** len(list_of_nums)
            result += power_of
        if result == i:
            answer_arr.append(result)
    return answer_arr

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?

function findArmstrongNumbers(arr) {
    let answer = []
    
    for (let i = 0; i < arr.length; i++) {
        let splittedArr = arr[i].toString().split('')
        let numLength = splittedArr.length
        let armstronNum = 0

        for (let  j = 0; j < splittedArr.length; j++)
            {
                armstronNum += Number(splittedArr[j]) ** numLength
            }
        if (armstronNum == arr[i]) {
            answer.push(armstronNum)
        }
    }
    return answer
};

module.exports = {findArmstrongNumbers}

